request-manager/graph/graph.go

```diff
@@ -155,51 +158,22 @@ func (g *Graph) PrintDot() {
 	fmt.Println("}")
 }
 
-// Returns true if a matches b, regardless of ordering
-func SlicesMatch(a, b []string) bool {
-	if a == nil && b == nil {
-		return true
-	}
-
-	if a == nil || b == nil {
-		return false
-	}
-
-	if len(a) != len(b) {
-		return false
-	}
-
-	for i, _ := range a {
-		ok := false
-		for j, _ := range b {
-			if a[i] == b[j] {
-				ok = true
-			}
-		}
-		if !ok {
-			return false
-		}
-	}
-
-	return true
-}
-
 // --------------------------------------------------------------------------
 
-// returns true iff the graph has at least one cycle in it
+// hasCycles returns true iff the graph has at least one cycle in it.
 func (g *Graph) hasCycles() bool {
 	seen := map[string]*Node{g.Source.Id: g.Source}
 	return g.hasCyclesDFS(seen, g.Source)
 }
 
-// returns true iff every node is reachable from the start node, and every path
-// terminates at the end node
+// isConnected returns true iff every node is reachable from the start node, and
+// every path terminates at the end node.
 func (g *Graph) isConnected() bool {
 	// Check forwards connectivity and backwards connectivity
 	return g.connectedToLastNodeDFS(g.Source) && g.connectedToFirstNodeDFS(g.Sink)
 }
 
-// Returns true if the last node in g is reachable from n
+// connectedToLastNodeDFS returns true iff the last node in g is reachable from n.
 func (g *Graph) connectedToLastNodeDFS(n *Node) bool {
```

Comment:
Can drop the "DFS" part of all func names. In a case like this, best practice is not tightly coupling func to implementation/algorithm. In other cases, e.g. when using a Strategy pattern, then maybe yes it makes sense to have the various implementations named after themselves.